WordPress/WordPressUITests/Tests/ReaderTests.swift

```diff
@@ -0,0 +1,36 @@
+import UITestsFoundation
+import XCTest
+
+class ReaderTests: XCTestCase {
+    private var readerScreen: ReaderScreen!
+
+    override func setUpWithError() throws {
+        setUpTestSuite()
+
+        _ = try LoginFlow.loginIfNeeded(siteUrl: WPUITestCredentials.testWPcomSiteAddress, email: WPUITestCredentials.testWPcomUserEmail, password: WPUITestCredentials.testWPcomPassword)
+        readerScreen = try EditorFlow
+            .goToMySiteScreen()
+            .tabBar.goToReaderScreen()
+    }
+
+    override func tearDownWithError() throws {
+        takeScreenshotOfFailedTest()
+        if readerScreen != nil && !TabNavComponent.isVisible() {
+            readerScreen.dismissPost()
+        }
+        try LoginFlow.logoutIfNeeded()
+        try super.tearDownWithError()
```

Comment:
It should not be necessary to call `super` here. Did you add this because of some error in the tests?

Reply:
No, I have just followed what all the other tests were doing. I did try removing the `super` now but it went into a recursive call loop. It runs just fine removing the whole line though.

Reply:
Ouch! My suggestions was not clear, sorry 😳 

"to call `super`" should have been "to call `super.tearDownWithError()` or "to call the super implementation". 

If you updated to 

```diff
-         try super.tearDownWithError()
+        try tearDownWithError()
```

it will indeed got into an infinite loop. ♻️ 

> No, I have just followed what all the other tests were doing.

Fair enough. Those methods are (should be?) empty in `XCTestCase` so it doesn't hurt to call them. It simply adds unnecessary code to the test.

Feel free to ignore this. I plan to audit the test suite for this kind of stuff during my HACK week. 👍 https://github.com/wordpress-mobile/WordPress-iOS/issues/17569